source/blog/_posts/2016-09-05-psh-managing-build-complexity.md

```diff
@@ -0,0 +1,290 @@
+by sharing a few sh scripts and using them to ensure that we could all work - it simply is the most natural format, you type a statement, execute it, verify it worked and copy and 
+paste it into a `script.sh` file. Everybody can do it and so it no longer was solely my task to create the environment we would all work in. Times were great! 
+
+But over the process of a one or two weeks some problems started to arise. We had gained a `*.dist` file, Docker had a different MySQL host then local development which meant copying whole 
+scripts just to change one or two lines. And worst of all: **If a statement at the beginning of shell script fails execution continues** - something that you will notice far too late.
+
+So what now? ANT? NO! [PHING](https://www.phing.info/) NO!! [Deployer](http://deployer.org/)? Maybe... Wait a minute... Deployer, although it looks very nice on first glance imports another concept from ANT that we never wanted to use again - [**Group tasks**](http://deployer.org/docs/tasks).
+Tasks, or in ANT lingo [Targets](https://ant.apache.org/manual/targets.html), to me look exactly like the reason why most ANT scripts simply can't be understood in the first place. As I described in my last post, it is extremely hard to get them right,
+and it is even harder to understand the order of things.
+
+Simply put, we did not want to loose the our current bare bones simple process in favor of another tool in between. Further we are all PHP developers, so requiring any tool not written in
+PHP would always mean that someone had problems installing *(A ANT and ARCH Linux story :) )* it. So I decided to create a thin wrapper, that would require minimal syntax adjustments
+and just remove the problems altogether.
+
+### PSH - The PHP Shell Helper
+
+And this is exactly what I did, and we are pleased enough with the solution that we even released it as [open source on Github](https://github.com/shopwareLabs/psh). 
+So let me give you a small introduction to it:
+ 
+##### Step by step initial setup
+ 
+Let's assume you have the following script: `scripts/install.sh`
+  
+```
+#!/usr/bin/env bash
+
+composer install
+
+ant -f shopware/build/build.xml -Ddb.user=sw -Ddb.password=sw -Ddb.host=localhost -Ddb.name=sw build-unit
+
+ln -s ../../../components/BackendAuth/SwagB2bBackendAuthPlugin shopware/custom/plugins/SwagB2bBackendAuthPlugin
+
+shopware/bin/console sw:plugin:refresh
+shopware/bin/console sw:plugin:install SwagB2bBackendAuthPlugin
+shopware/bin/console sw:plugin:activate SwagB2bBackendAuthPlugin
+```
+
+Easy to read, isn't it? Install all composer dependencies, install shopware, link a plugin to the shopware plugin directory and install it. You run it, it works.
+The second time - although you see an error that the symlink already exists - it works too. To prevent the error you can try to delete the symlink again, which would update the script to this:
```

Comment:
Alternatively use the overwrite argumnt for `ls`: 
`ln -sf ../../../components/BackendAuth/SwagB2bBackendAuthPlugin shopware/custom/plugins/SwagB2bBackendAuthPlugin`